### PR TITLE
bugfix/payment requests & deeplinks

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

We were using the parsed decimal number instead of the hex number to get the token amount from the `data`

https://trello.com/c/qhpAxjC1/192-walletconnect-not-working-on-android and https://trello.com/c/ubVPsajC/193-payment-request-deeplink-showing-wrong-amount

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

**Review by rickycodes — APPROVED** [review]

File: app/components/Views/Entry/index.js [hunk 1]
```diff
@@ -106,8 +106,9 @@ class Entry extends PureComponent {
 		}
 		const deeplink = params['+non_branch_link'] || uri || null;
 		if (deeplink) {
-			const existingUser = await AsyncStorage.getItem('@MetaMask:existingUser');
-			!existingUser ? DeeplinkManager.setDeeplink(deeplink) : DeeplinkManager.parse(deeplink);
+			const { KeyringController } = Engine.context;
+			const isUnlocked = KeyringController.isUnlocked();
+			isUnlocked ? DeeplinkManager.parse(deeplink) : DeeplinkManager.setDeeplink(deeplink);
```
Comment (estebanmino) [inline app/components/Views/Entry/index.js:111]: we're checking if keyring is unlocked to store a deeplink or parse it right away

File: app/components/Views/Send/index.js [hunk 2]
```diff
@@ -292,7 +291,7 @@ class Send extends PureComponent {
 						amount: BNToHex(tokenAmount)
 					}),
 					value: '0x0',
-					readableValue: parameters.uint256 || '0'
+					readableValue: fromTokenMinimalUnit(parameters.uint256, selectedAsset.decimals) || '0'
```
Comment (estebanmino) [inline app/components/Views/Send/index.js:294]: now we have to handle token "atomic" units from deeplinks

File: app/components/Views/SendFlow/Confirm/index.js [hunk 3]
```diff
@@ -436,7 +436,7 @@ class Confirm extends PureComponent {
 				contractBalances[address] ? contractBalances[address] : '0',
 				decimals
 			)} ${symbol}`;
-			[transactionTo, , amount] = decodeTransferData('transfer', data);
+			[transactionTo, amount] = decodeTransferData('transfer', data);
```
Comment (estebanmino) [inline app/components/Views/SendFlow/Confirm/index.js:439]: we were taking a parsed number from the hex format, now we take the hex number

**Review by ibrahimtaveras00 — APPROVED** [review]
QA Passed 👍

**Review by rickycodes — APPROVED** [review]

**Pull request merged**